__tests__/app.test.js

```diff
@@ -113,14 +113,98 @@ describe('GET /api/articles', () => {
+        expect(typeof article.votes).toBe('number');
+        expect(typeof article.article_img_url).toBe('string');
+        expect(typeof article.comment_count).toBe('number');
+      });
+    });
+  });
+  test('GET:200 sends all articles when no topic is specified with correct data', () => {
+    return request(app).get('/api/articles').expect(200).then(({ body }) => {
+      expect(body.articles[0]).toHaveProperty('article_id', 3);
+      expect(body.articles[0]).toHaveProperty('title', 'Eight pug gifs that remind me of mitch');
+      expect(body.articles[0]).toHaveProperty('author', 'icellusedkars');
+      expect(body.articles[0]).toHaveProperty('topic', 'mitch');
+      expect(body.articles[0]).toHaveProperty('created_at', '2020-11-03T09:12:00.000Z');
+      expect(body.articles[0]).toHaveProperty('votes', 0);
+      expect(body.articles[0])
+        .toHaveProperty('article_img_url', 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700');
+      expect(body.articles[0]).toHaveProperty('comment_count', 2);
+    });
+  });
+  test('GET:200 sends all filtered articles when a topic is specified in the correct order', async () => {
+    const topic = 'mitch';
+    return request(app).get(`/api/articles?topic=${topic}`).expect(200).then(({ body }) => {
+      expect(body.articles).toBeInstanceOf(Array);
+      expect(body.articles).toBeSortedBy('created_at', {
+        descending: true,
+      });
+      expect(body.articles.length === 12).toBe(true);
+      body.articles.forEach((article) => {
+        expect(article).toHaveProperty('topic', 'mitch');
+      });
+    });
+  });
+  test('GET:200 sends all filtered articles when a topic is specified in ascending order', async () => {
+    const topic = 'mitch';
+    return request(app).get(`/api/articles?topic=${topic}&order=ASC`).expect(200).then(({ body }) => {
+      expect(body.articles).toBeInstanceOf(Array);
+      expect(body.articles).toBeSortedBy('created_at');
+      expect(body.articles.length === 12).toBe(true);
+      body.articles.forEach((article) => {
+        expect(article).toHaveProperty('topic', 'mitch');
+      });
+    });
+  });
+  test('GET:200 sends all filtered articles when a topic is specified in ascending order should be sorted by \'title\' column', async () => {
+    const topic = 'mitch';
+    const sort_by = 'title';
+    return request(app).get(`/api/articles?topic=${topic}&order=ASC&sort_by=${sort_by}`).expect(200).then(({ body }) => {
+      expect(body.articles).toBeInstanceOf(Array);
+      expect(body.articles).toBeSortedBy('title');
+      expect(body.articles.length === 12).toBe(true);
+      body.articles.forEach((article) => {
+        expect(article).toHaveProperty('topic', 'mitch');
+      });
+    });
+  });
+  test('GET:400 sends an appropriate status and error message when given an invalid endpoint', () => {
     return request(app)
       .get('/api/articles/not-right-path')
```

Comment:
_technically_ 'not-right-path' could be a topic! (It would be a very weird topic, but it _could_ be one!)